### PR TITLE
2D Tile Painter 

### PR DESCRIPTION
**Pull request by alexjusung1**
## Description

Adds scripts for a prototype editor tool. While the editorOn boolean is switched on (from the Editor Panel) and the SceneManager object is selected, the editor reads for mouse input.
When the left mouse button is clicked, the editor searches for a PaperSquare object at the position and saves to a variable (copyFrom). If it is not found, it creates a new PaperSquare object by cloning the PaperSquare from the variable.
When the right mouse button is clicked, the editor deletes the PaperSquare object at the position if it is present.

## Changed Scipts

Edited:
- Scripts/Foldable Paper/PaperSquare.cs: minor addition of 2 public variables and OnValidate method (to update positions of topHalf and bottomHalf)

Added:
- Scripts/Test/PaperSquares.cs: Parent for PaperSquare instances, managing their creation/deletion and location in Scene
- Scripts/Test/SceneManager.cs, SceneEditor.cs: Attached to a Quad object, used to read input while the Quad object is selected in the editor

## Changed Files

- BaseTile Prefab: Test prefab until we have something concrete to work with
- ClearMaterial.mat: Material of Quad (so that SceneEditor can read input)

## Related Task

TPT 1

## How Has This Been Tested?

- Tested adding tiles to previously deleted tiles
- Tested deleting the origin tile
- Tested adding/deleting tiles outside of the bounds of the editor
- Tested deleting the PaperSquare stored in the copyFrom variable (currently set to point back to origin square)

TODO:
- Create a tree for the copyFrom variable (so that it rolls back to the previous PaperSquare if it is not present)
- Use EditorTool API instead of what I have rn (https://docs.unity3d.com/ScriptReference/EditorTools.EditorTool.html)
- Expand to 3D
- Implement Undo for adding/removing PaperSquares (or at least flag isDirty)

### CODE REVIEW

File: Barkane/.vsconfig [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿{
+  "version": "1.0",
+  "components": [
+    "Microsoft.VisualStudio.Workload.ManagedGame"
```
Comment (alanrtao) [inline Barkane/.vsconfig:4]: Could you delete this file? I have it ignored on my end although not really sure if it has any side effects.

File: Barkane/Assets/Scripts/Test/SceneManager.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    [SerializeField] private bool editorOn = false;
+    public bool EditorOn => editorOn;
+    [SerializeField] private PaperSquares squares;
+    public PaperSquares Squares => squares;
+    [SerializeField] private Plane editPlane;
+
+    // Start is called before the first frame update
+    private void Start()
+    {
+        editPlane = new Plane(Vector3.up, Vector3.zero);
```
Comment (alanrtao) [inline Barkane/Assets/Scripts/Test/SceneManager.cs:17]: you can move this to variable initialization and removev Start

File: Barkane/Assets/Scripts/Test/SceneManager.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        if (selectPos == null)
+        {
+            return;
+        }
+        squares.RemoveSquare((Vector3)selectPos);
+    }
+
+    private void OnMouseDown()
+    {
+        print("asdfasdf");
```
Comment (alanrtao) [inline Barkane/Assets/Scripts/Test/SceneManager.cs:52]: remove

File: Barkane/ProjectSettings/Packages/com.unity.probuilder/Settings.json [hunk 4]
```diff
@@ -37,6 +37,11 @@
                 "type": "System.Boolean, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089",
                 "key": "experimental.enabled",
                 "value": "{\"m_Value\":false}"
+            },
+            {
```
Comment (alanrtao) [inline Barkane/ProjectSettings/Packages/com.unity.probuilder/Settings.json:41]: remove these changes. sorry this should be in gitignore, but we are making the gitignore change on main branch, so just removing the changes should be good enough

**Review by alanrtao — CHANGES_REQUESTED** [review]
Good work! Just some minor things with git and cleaning up code

File: Barkane/.vsconfig (hunk 1 above)
Comment (alexjusung1) [reply Barkane/.vsconfig:4]: Got it; it's just a Visual Studio config file. We should add it to gitignore if it's not necessary

**Pull request closed without merging**